scalyr_agent/log_processing.py

```diff
@@ -1216,18 +1216,40 @@ def __find_copy_truncate_file(self):
         @return Most recent file that matches copy truncate filename heuristic
         """
         dir_path = os.path.dirname(self.__path)
-        file_prefix = os.path.basename(self.__path)
-        file_prefix = file_prefix[:-4] if file_prefix.endswith(".log") else file_prefix
+        file_name = os.path.basename(self.__path)
+        file_prefix, file_ext = os.path.splitext(file_name)
 
         # Files starting with the file prefix with create time within the last 5 minutes
         # gzip files are excluded
         # Use file with most recent creation date when there are multiple files
+
         possible_copy_filepaths = list(
             six.moves.filter(
-                lambda f: os.path.basename(f).startswith(file_prefix)
-                and not f.endswith(".gz")
+                lambda f: not f.endswith(".gz")
                 and f != self.__path
-                and (int(time.time()) - self.__file_system.stat(f).st_ctime < 300),
+                and (int(time.time()) - self.__file_system.stat(f).st_ctime < 300)
+                # Since it's not a trivial task to handle all possible variants of the logrotate configuration,
```

Comment:
IIRC, we already have some docs on log rotate, but if not there already, it would be good to also add example file names and formats which are supported.